admin/services/redis-cache.service.js

```diff
@@ -0,0 +1,213 @@
+        if (thisTables) {
+          tables = tables.concat(thisTables)
+        }
+      })
+    }
+    if (tables.length) {
+      const stream = redis.scanStream()
+      stream.on('data', keys => {
+        const keyRegex = new RegExp(`(_|-)(${tables.join('|')})`, 'i')
+        const matchedKeys = keys.filter(k => keyRegex.test(k))
+        if (matchedKeys.length) {
+          const pipeline = redis.pipeline()
+          matchedKeys.forEach(key => {
+            console.log(`REDIS (dropAffectedCaches): Dropped \`${key}\``)
+            pipeline.del(key)
+          })
+          pipeline.exec()
+        }
+      })
+      stream.on('end', resolve)
+    } else {
+      resolve(true)
+    }
+  })
+}
+
+/**
+ * Find a Redis key based on a Regex pattern
+ * @param pattern - regex pattern
+ * @returns {Array}
+ */
+const findKeys = pattern => {
+  return new Promise((resolve, reject) => {
+    let foundKeys = []
+    const stream = redis.scanStream()
+    stream.on('data', keys => {
+      foundKeys = foundKeys.concat(keys.filter(key => pattern.test(key)))
+    })
+    stream.on('end', () => {
+      resolve(foundKeys)
+    })
+  })
+}
+
+/**
+ * Manually updates data in Redis and then sends to the Azure `sql-update` queue,
+ * which will be performed in SQL Server by `/functions`
+ * @param serviceKey - the prefix of the key in the format `serviceName.methodName`
+ * @param changes - object with changes in the format { tableName: 'foo-bar', update: { 1: { foo: 'bar' } } }
+ * @returns {Boolean}
+ */
+redisCacheService.update = (serviceKey, changes) => {
+  return new Promise(async (resolve, reject) => {
+    if (!REDIS_CACHING) {
+      resolve(false)
+    } else {
+      const keys = await findKeys(new RegExp(`^${serviceKey}_`))
+      const foundKey = keys.length ? keys[0] : false
+      if (!foundKey) {
+        resolve(false)
+      } else {
+        redis.get(foundKey, (err, result) => {
+          if (err || !result) {
+            resolve(false)
+          } else {
+            result = JSON.parse(result)
+            result.recordset = result.recordset.map(r => {
+              if (changes.update && changes.update[r.id]) {
+                for (let prop in changes.update[r.id]) {
+                  r[prop] = changes.update[r.id][prop]
+                }
+              }
+              if (changes.delete && changes.delete.indexOf(r.id.toString()) > -1) {
+                return false
+              }
+              return r
+            }).filter(r => r !== false)
+            redis.set(foundKey, JSON.stringify(result), async () => {
+              console.log(`REDIS (update): Updated \`${foundKey}\``)
+              const sqlUpdateQueueName = queueNameService.getName(queueNameService.NAMES.SQL_UPDATE)
+              await azureQueueService.addMessageAsync(sqlUpdateQueueName, { version: 2, messages: [changes] })
+              console.log(`REDIS (update): Sent \`${foundKey}\` update to \`sql-update\` message queue`)
```

Comment:
introduce `models/logger` and use logger.debug` in every instance where `console.log` is used

Reply:
Changed all to use the logger.